### DISCUSSION TITLE
Problem with Speaker Diarization

### DISCUSSION THREAD

**Discussion by Atefeh197** [NVIDIA-NeMo/NeMo#6415] [General]
Hello

I want to do Speaker Diarization with NeMo. I use [this](https://colab.research.google.com/github/NVIDIA/NeMo/blob/stable/tutorials/speaker_tasks/ASR_with_SpeakerDiarization.ipynb#scrollTo=PQ2SKvFykXyg) but I got error in this line:

word_hyp, word_ts_hyp = asr_decoder_ts.run_ASR(asr_model)

it is : TypeError: _get_batch_preds() takes 1 positional argument but 2 were given

how I can fix it?

**Comment by Rumeysakeskin** [comment]
@Atefeh197, the code works without error. Could you restart "run all" again in the "runtime" section.
